test/list/main.go

```diff
@@ -24,23 +24,75 @@ import (
 	"go/ast"
 	"go/parser"
 	"go/token"
+	"io/ioutil"
 	"log"
 	"os"
 	"path/filepath"
+	"regexp"
 	"strconv"
 	"strings"
 )
 
 var (
-	dumpTree = flag.Bool("dump", false, "print AST")
-	dumpJson = flag.Bool("json", false, "output test list as JSON")
-	warn     = flag.Bool("warn", false, "print warnings")
+	dumpTree            = flag.Bool("dump", false, "print AST")
+	dumpJson            = flag.Bool("json", false, "output test list as JSON")
+	warn                = flag.Bool("warn", false, "print warnings")
+	writeConformanceDoc = flag.Bool("conformance", false, "write a conformance document")
 )
 
 type Test struct {
 	Loc      string
 	Name     string
 	TestName string
+	Comment  string
+}
+
+type ConformanceData struct {
+	// A URL to the line of code in the kube src repo for the test
+	URL string
+	// Extracted from the "Testname:" comment before the test
+	TestName string
+	// Extracted from the "Description:" comment before the test
+	Description string
+}
+
+func convertToConformanceData(test *Test) ConformanceData {
+	baseURL := "https://github.com/kubernetes/kubernetes/tree/master/"
```

Comment:
This may break the code path for branched out code like `1.9`, `1.x` etc...

Reply:
are you referring to theURL path being different?\

Reply:
What I meant is -  let's say if I run this tool from `1.10` branch in future this will point me to code master tree which is wrong, so we should have some other mechanism to find out current branch and update the baseURL accordingly.

Reply:
Is there a reliable way for the tool to know what branch its on? Even if it asked git for the current branch it could be some user-defined branch which isn't helpful.  I guess we could allow for a cmd-line flag to specify the version/branch but default to "master".  @mkulke thoughts?